src/loadElmStories.tsx

```diff
@@ -25,15 +26,19 @@ export const loadElmStories = (
   name: string,
   module: NodeModule,
   elmApp: { Elm: { Main: any } },
-  storyNames: string[]
+  storyNames: string[],
+  ports: (ports: unknown) => void | null
 ) => {
   if (!elmApp.Elm.Main) {
     throw new Error("Elm storybook module did not exist with name `Main`")
   }
   const stories = storiesOf(name, module)
   for (const storyName of storyNames) {
-    stories.add(storyName, () => (
-      <ElmComponent src={elmApp.Elm.Main} flags={storyName} />
-    ))
+    stories.add(storyName, () => {
+      if (ports) {
+        return <ElmComponent src={elmApp.Elm.Main} flags={storyName} ports={ports} />
+      }
+      return <ElmComponent src={elmApp.Elm.Main} flags={storyName} />
```

Comment:
If ports is optional, then there's no harm in always passing it through (if it's undefined it's the same as not specifying the property at all).

```suggestion
      return <ElmComponent src={elmApp.Elm.Main} flags={storyName} ports={ports} />
```